fetch.bs

```diff
@@ -4303,14 +4331,21 @@ Range Requests</cite>. [[HTTP-RANGE]] However, this is not widely supported by b
     <var>httpRequest</var>'s
     <a for=request>credentials mode</a>.
 
+   <li><p>Set <var>httpCache</var> to the result of
+   <a lt="determine the HTTP cache partition">determining the HTTP cache partition</a> given
+   <var>httpRequest</var>.</p></li>
+
+   <li><p>If <var>httpCache</var> is null, then set <var>httpRequest</var>'s
+   <a for=request>cache mode</a> to "<code>no-store</code>".</p></li>
```

Comment:
How common will this be? Note that currently it can have the effect of disabling upstream caches as well; see #722.

Reply:
Hmm that's a good point. It should basically never be null, but I think this is buggy for user-initiated top-level navigations at the moment as they don't have a client. (But that might be buggy in Fetch overall...)

Reply:
I put this null handling here to handle the case where "determine the HTTP cache partition" returns null, which it does if request's [client](https://fetch.spec.whatwg.org/#concept-request-client) is null. Not sure when that happens, though.

Reply:
One example is a user-initiated top-level navigation. For those we should use the site of the request URL, but this would not do that.

I suspect the client is also wrong if tab A navigates tab B. We wouldn't want to use the key from tab A.